### PR TITLE
Fix BBL to build without a sysroot

### PR DESCRIPTION
**Pull request by arichardson**
This also fixes mcall_trap to return SBI_ERR_NOT_SUPPORTED (-2) instead of
the platform dependent ENOSYS value.

### CODE REVIEW

File: util/string.c [hunk 1]
```diff
@@ -1,15 +1,15 @@
 // See LICENSE for license details.
 
-#include <string.h>
+#include "string.h"
 #include <stdint.h>
-#include <ctype.h>
+// #include <ctype.h>
```
Comment (arichardson) [inline util/string.c]:
```suggestion
```

**Comment by jrtc27** [comment]
You still need a sysroot to configure it.

**Comment by arichardson** [comment]
Seemed to work without one for me.

**Comment by arichardson** [comment]
But anyway it should not be using errno.h from some random sysroot.

**Comment by arichardson** [comment]
> You still need a sysroot to configure it.

I had a stale sysroot lying around, configure fails now.

**Comment by arichardson** [comment]
Okay it works now after `self.CFLAGS.extend(["-nostartfiles", "-nostdlib", "-static", "-ffreestanding"])`

**Pull request merged**